crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1388,3 +1388,22 @@ unsafe impl<Q: WorldQuery> WorldQuery for NopWorldQuery<Q> {
 
 /// SAFETY: `NopFetch` never accesses any data
 unsafe impl<Q: WorldQuery> ReadOnlyWorldQuery for NopWorldQuery<Q> {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate as bevy_ecs;
+
+    // regression test for https://github.com/bevyengine/bevy/issues/8010.
```

Comment:
I'd really like it if this was moved into a discrete test with a name. It's tricky to grasp the structure here, and seems easy to revert out of confusion.

Reply:
Fair enough, I'm just not aware of any "runnable" code I could include in a test that would be relevant here.

Reply:
Hmm. You should be able to define types within functions just fine, based on a quick playground experiment. `assert_is_system` may also be nice to toss on though.